phpcr-shell/querying.rst

```diff
@@ -169,6 +169,40 @@ Arguments:
   property
 - **value**: Value to append
 
+expr
+""""
+
+Evaluate an expression. This function is very powerful in that it enables
+you to use the `Symfony Expression Language`_ to evaluate an expression, the
+result of which can be assigned to a property.
+
+Within the expression you have access to the `row` object
+(`RowInterface`_).
+
+Set the value of ``a.title`` to the node name:
+
+.. code-block:: bash
+
+    PHPCRSH> UPDATE [nt:unstructured] AS a SET a.title = expr('row.getNode().getName()')
+
+Set the value of ``a.title`` to the value of the property :
+
+.. code-block:: bash
+
+    PHPCRSH> UPDATE [nt:unstructured] AS a SET a.title = expr('row.getNode().getName()')
+
+Increment the value of a property:
+
+.. code-block:: bash
+
+    PHPCRSH> UPDATE [nt:unstructured] AS a SET a.count = expr('row.getNode().getPropertyValue("count") + 1')
+
+Set the value of ``a.title`` from the value of a joined node:
+
+.. code-block:: bash
+
+    PHPCRSH> UPDATE [nt:unstructured] AS a INNER JOIN [nt:something] AS b ON a.foo = b.bar SET a.title = expr('row.getNode("b").getPropertyValue("something")')
```

Comment:
hey, this is indeed really powerful!

maybe we should add a note or warning that these things are executed in userland and thus should not be assumed to be as efficient as sql update statements. and if the result set is too large, we will run into memory problems as jackalope never discards loaded nodes.